src/loaders/tvg/tvgTvgLoadParser.cpp

```diff
@@ -20,76 +20,85 @@
  * SOFTWARE.
  */
 
-#include <string.h>
+#include <memory.h>
 #include "tvgTvgLoadParser.h"
 
+
+/************************************************************************/
+/* Internal Class Implementation                                        */
+/************************************************************************/
+
 enum class LoaderResult { InvalidType, Success, SizeCorruption, MemoryCorruption, LogicalCorruption };
 
 static LoaderResult _parsePaint(tvgBlock block, Paint ** paint);
 
-static tvgBlock _readTvgBlock(const char *pointer)
+static tvgBlock _readTvgBlock(const char *ptr)
 {
     tvgBlock block;
-    block.type = *pointer;
-    _read_tvg_ui32(&block.length, pointer + TVG_INDICATOR_SIZE);
-    block.data = pointer + TVG_INDICATOR_SIZE + BYTE_COUNTER_SIZE;
-    block.blockEnd = block.data + block.length;
+    block.type = *ptr;
+    _read_tvg_ui32(&block.length, ptr + TVG_INDICATOR_SIZE);
+    block.data = ptr + TVG_INDICATOR_SIZE + BYTE_COUNTER_SIZE;
+    block.end = block.data + block.length;
     return block;
 }
 
-static bool _readTvgHeader(const char **pointer)
+static bool _readTvgHeader(const char **ptr)
 {
-    // Sign phase, always "TVG" declared in TVG_HEADER_TVG_SIGN_CODE
-    if (memcmp(*pointer, TVG_HEADER_TVG_SIGN_CODE, TVG_HEADER_TVG_SIGN_CODE_LENGTH)) return false;
-    *pointer += TVG_HEADER_TVG_SIGN_CODE_LENGTH; // move after sing code
+    //Sign phase, always TVG_BIN_HEADER_SIGNATURE is declared
+    if (memcmp(*ptr, TVG_BIN_HEADER_SIGNATURE, TVG_BIN_HEADER_SIGNATURE_LENGTH)) return false;
+    *ptr += TVG_BIN_HEADER_SIGNATURE_LENGTH;
 
-    // Standard version number, declared in TVG_HEADER_TVG_VERSION_CODE
-    if (memcmp(*pointer, TVG_HEADER_TVG_VERSION_CODE, TVG_HEADER_TVG_VERSION_CODE_LENGTH)) return false;
-    *pointer += TVG_HEADER_TVG_VERSION_CODE_LENGTH; // move after version code
+    //Version number, declared in TVG_BIN_HEADER_VERSION
+    if (memcmp(*ptr, TVG_BIN_HEADER_VERSION, TVG_BIN_HEADER_VERSION_LENGTH)) return false;
+    *ptr += TVG_BIN_HEADER_VERSION_LENGTH;
 
-    // Matadata phase
-    uint16_t meta_length; // Matadata phase length
-    _read_tvg_ui16(&meta_length, *pointer);
-    *pointer += 2; // move after length
+    //Mata data for proof?
```

Comment:
mata -> meta

Reply:
i guess meta sector will be removed. leave it for now.